test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(inputCommand.equals("add")||inputCommand.equals("clear")||inputCommand.equals("delete")){
```

Comment:
Perhaps another solution would be to make use of the `isMutating()` that you have created.

```java
if(new Parser().parseCommand(inputCommand).isMutating()) {
    assertEquals(addressBook, saveFile.load());
}
```

In this way you ensure that when a new mutating command is added, you can be sure that you are testing those new commands.